tests/conftest.py

```
@@ -73,8 +92,203 @@ def _override_get_db():
+            Path(path.source_path).mkdir(exist_ok=True, parents=True)
+        except PermissionError:
+            # If we can't create it, we assume the test will handle it or use a different path
+            pass
+        return path
+
+    return _factory
+
+@pytest.fixture
+def file_inventory_factory(db_session: Session, monitored_path_factory):
+    """Factory for FileInventory objects."""
+    def _factory(path_id: int = None, file_path: str = None):
```

Ruff (0.15.1)

[warning] Missing return type annotation for private function `_factory`

(ANN202)